### PR TITLE
Refactor library model & library import to add fields and rename as collection

### PR DESCRIPTION
**Pull request by rlskoeser**

### CODE REVIEW

File: geniza/corpus/models.py
Comment (thatbudakguy) [inline geniza/corpus/models.py:10]: `Collection.collection` feels weird to me, but I'm guessing there's a reason it's not `Collection.name`?

File: geniza/corpus/models.py
Comment (thatbudakguy) [inline geniza/corpus/models.py:12]: idk what we're using this for -  if it's just display, `CharField` seems fine, but if we want to do anything with the actual info later we might wanna store real coords or geonames or something

File: geniza/corpus/models.py
Comment (thatbudakguy) [inline geniza/corpus/models.py:37]: whoa, I've never seen this `constraints` API before! since you defined uniqueness here, should be trivial to also implement `get_natural_key()` – just a thought

**Review by thatbudakguy — APPROVED** [review]
the changes for `Collection` are a little strange to me but i'm sure they're based on convos with the team so it's just context I don't have; mentally going off of PEMM but that might not map to this model. would different collections at the same library have the the same abbrev?

File: geniza/corpus/models.py
Comment (rlskoeser) [reply geniza/corpus/models.py:10]: That's a good point; but name also seems a little weird, since this field is optional

File: geniza/corpus/models.py
Comment (rlskoeser) [reply geniza/corpus/models.py:12]: I thought about that too, and mentioned briefly in the working meeting when we discussed these changes; but I'm recommending we hold off on that until we have a use case, since our priority is getting the spreadsheet data migrated into the database.

File: geniza/corpus/models.py
Comment (rlskoeser) [reply geniza/corpus/models.py:37]: This was new to me too! Credit to @kmcelwee for this one, I would have just used `unique_together`, but apparently that's deprecated now and might go away eventually.

**Comment by rlskoeser** [comment]
> the changes for `Collection` are a little strange to me but i'm sure they're based on convos with the team so it's just context I don't have; mentally going off of PEMM but that might not map to this model. would different collections at the same library have the the same abbrev?

Sounds like the changes aren't documented as well as they should be on the GitHub issue. Sorry about that.

We needed some adjustments to handle some outliers, including:
- multiple collections at the same institution
- one collection shared across two institutions

I asked about the abbreviation — it seems to be the collection abbreviation in some cases and the institution abbreviation in others (example: ENA vs JTS); I think that's ok as long as abbreviations are unique.  

Any thoughts or lessons learned from the PEMM collections that are relevant here?

**Comment by rlskoeser** [comment]
@thatbudakguy added natural foreign keys for collection and languagescript; thanks for the suggestion. Anything else you think we should address before merging & delivering for testing?

**Review by thatbudakguy — APPROVED** [review]
let's merge it!

**Pull request merged**
